modules/server/src/mapping/createConnectionResolvers.ts

```diff
@@ -24,35 +26,73 @@ const createConnectionResolvers: CreateConnectionResolversFn = ({
 	getServerSideFilter,
 	Parallel,
 	type,
-}) => ({
-	[type.name]: {
-		aggregations: resolveAggregations({ type, getServerSideFilter }),
-		configs: async (parentObj, { fieldNames }: { fieldNames: string[] }) => {
-			return {
-				downloads: type.config?.[ConfigProperties.DOWNLOADS],
-				extended: fieldNames
-					? type.extendedFields.filter((extendedField: ExtendedConfigsInterface) =>
-							fieldNames.includes(extendedField.fieldName),
-					  )
-					: type.extendedFields,
-				...(createStateResolvers && {
-					facets: type.config?.[ConfigProperties.FACETS],
-					matchbox: type.config?.[ConfigProperties.MATCHBOX],
-					table: type.config?.[ConfigProperties.TABLE],
-				}),
-			};
+}) => {
+	const configs = async (parentObj, { fieldNames }: { fieldNames: string[] }) => {
+		return {
+			downloads: type.config?.[ConfigProperties.DOWNLOADS],
+			extended: fieldNames
+				? type.extendedFields.filter((extendedField: ExtendedConfigsInterface) =>
+						fieldNames.includes(extendedField.fieldName),
+				  )
+				: type.extendedFields,
+			...(createStateResolvers && {
+				facets: type.config?.[ConfigProperties.FACETS],
+				matchbox: type.config?.[ConfigProperties.MATCHBOX],
+				table: type.config?.[ConfigProperties.TABLE],
+			}),
+		};
+	};
+
+	// TODO: memoise instead of context
+	// just same request really - maybe  JSON.stringify
+	const aggregationsQuery = resolveAggregations({ type, getServerSideFilter });
+	const aggregationsResolver = async (obj, args, context, info) => {
+		const aggs = await aggregationsQuery(obj, args, context, info);
+		return aggregationsToGraphql(aggs);
+	};
+
+	const defaultHitsResolver = resolveHits({ type, Parallel, getServerSideFilter });
+	const hitsResolver = enableDocumentHits
+		? defaultHitsResolver
+		: async (obj, args, context, info) => {
+				/*
+				 * Checks for aggregations field in full query and retrieves args
+				 * Popular parsing `info` libs do not include these operations properties
+				 */
+				const typeNameConnectionProperty = info.operation.selectionSet.selections[0];
+				const isAggregationsQueried = typeNameConnectionProperty.selectionSet.selections.some(
```

Comment:
if `aggregations` field is queried also, pull out the variables and call the aggregation resolving code, then resolve hits from the result.

the aggregation resolver code can be memoized so it's only called once. better than sharing potentially huge objects over context